app/home/routes.py

```diff
@@ -91,4 +91,57 @@ def create_listing():
             img_file.save(file_path)
 
         return redirect(url_for('base_blueprint.route_default'))
-    return render_template('create_listing.html', form=form)
+    return render_template('create_property.html', form=form)
+
+
+@blueprint.route('/view-properties')
+def view_properties():
+    """
+    This function returns the view_properties.html the properties and images from the database.
+    Only one picture for each property is needed to be displayed on each property in the view_properties.html page.
+    So a dictionary is used to return each id(property_id) as a key and property image(image_name) as value.
+    """
+    properties = db.session.query(Property).order_by(Property.date.desc())
+    property_images = db.session.query(PropertyImage).order_by(PropertyImage.id.asc())
+
+    pictures = {}
+    for property_img in property_images:
+        pictures[property_img.property_id] = property_img.image_name
+
+    return render_template('view_properties.html',
+                           properties=properties,
+                           property_images=property_images,
+                           pictures=pictures)
+
+
+@blueprint.route('/<int:prop_id>/details')
```

Comment:
Login required for this route as well

So what it means is that you want this feature to be accessible only to authenticated users 
And not for it to be world readable